lib/supplejack_common/paginated_collection.rb

```diff
@@ -127,9 +128,20 @@ def next_paginated_url(url)
     def next_scroll_url(url)
       return url + joiner(url) + scroll_url_query_params unless klass._document.present?
 
-      base_url = url.match('(?<base_url>.+\/collection)')[:base_url]
+      case @scroll_type
+      when 'elasticsearch'
+        scroll_id = JSON.parse(klass._document.body)['_scroll_id']
+        base_url = url.match('(?<base_url>.+\/search)')[:base_url]
+        next_scroll_url = base_url + "/_search/scroll/#{scroll_id}?" + scroll_url_query_params
+      when 'tepapa'
+        base_url = url.match('(?<base_url>.+\/collection)')[:base_url]
+        next_scroll_url = base_url + klass._document.headers[:location] + joiner(url) + scroll_url_query_params
```

Comment:
Any chance we could implement this logic in the parser itself? Using a callback maybe. I feel bad about implementing some "tepapa" logic in the open source repo.

Reply:
Fair call I had thought this as well. I'll see if we can pass a block or something in the parser instead